src/unix/timer.c

```diff
@@ -71,7 +72,7 @@ int uv_timer_start(uv_timer_t* handle,
   if (uv__is_active(handle))
     uv_timer_stop(handle);
 
-  clamped_timeout = handle->loop->time + timeout;
+  clamped_timeout = (handle->timeout <= 0 ? handle->loop->time : handle->timeout) + timeout;
```

Comment:
I don't get what this is trying to do.  Use the previous `clamped_timeout` as the base?

Is it possible you simply forgot to call `uv_update_time()` before calling `uv_timer_start()`?